### PR TITLE
add e2e test case user deleteall

### PR DESCRIPTION
**Pull request by M-Buntoro**
### Summary

Add end to end test case for `mmctl user deleteall` 

### Ticket Link

https://mattermost.atlassian.net/browse/MM-29152
Fixes https://github.com/mattermost/mattermost-server/issues/15662

### CODE REVIEW

File: commands/user_e2e_test.go [hunk 1]
```diff
@@ -136,3 +136,59 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
+		s.Require().NotNil(err)
+		s.Len(printer.GetLines(), 0)
+		s.Len(printer.GetErrorLines(), 0)
+
+		// expect users not deleted
+		_, err = s.th.App.GetUser(s.th.BasicUser2.Id)
+		s.Require().Nil(err)
+	})
+
+	s.Run("Delete all user as system admin", func() {
```
Comment (isacikgoz) [inline commands/user_e2e_test.go:160]: Shall we add a couple of users here?

File: commands/user_e2e_test.go [hunk 2]
```diff
@@ -136,3 +136,59 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
+		cmd := &cobra.Command{}
+		confirm := true
+		cmd.Flags().BoolVar(&confirm, "confirm", confirm, "confirm")
+
+		err := deleteAllUsersCmdF(s.th.SystemAdminClient, cmd, []string{})
+		s.Require().NotNil(err)
+		s.Len(printer.GetLines(), 0)
+		s.Len(printer.GetErrorLines(), 0)
+
+		// expect users not deleted
```
Comment (isacikgoz) [inline commands/user_e2e_test.go:172]: I would expect to users to be deleted? Could this be related to `config.ServiceSettings.EnableAPIUserDeletion`?

File: commands/user_e2e_test.go [hunk 3]
```diff
@@ -136,3 +136,59 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
+		_, err = s.th.App.GetUser(s.th.BasicUser2.Id)
+		s.Require().Nil(err)
+	})
+
+	s.Run("Delete all user", func() {
+		printer.Clean()
```
Comment (isacikgoz) [inline commands/user_e2e_test.go:178]: Shall we add a couple of users here just in case?

**Review by isacikgoz — CHANGES_REQUESTED** [review]
Awesome @M-Buntoro 🚀  Thanks a lot for all of your contributions, I have spotted a few issues can you check please?

File: commands/user_e2e_test.go (hunk 1 above)
Comment (M-Buntoro) [reply commands/user_e2e_test.go:160]: cool, I'll add some user here

File: commands/user_e2e_test.go (hunk 3 above)
Comment (M-Buntoro) [reply commands/user_e2e_test.go:178]: cool, I'll add some user

File: commands/user_e2e_test.go (hunk 2 above)
Comment (M-Buntoro) [reply commands/user_e2e_test.go:172]: That's the strange part, I tried enabling it but still results in 404 when using system admin client, but the endpoint works only for local client. Not sure if the intended flow should be like that too

Comment (mgdelacroix) [reply commands/user_e2e_test.go:172]: Bear in mind that [this endpoint only works through local mode](https://api.mattermost.com/#tag/users/paths/~1users/delete) by design :slightly_smiling_face:

File: commands/user_e2e_test.go [hunk 4]
```diff
@@ -136,3 +136,64 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
+		s.Require().NotNil(err)
+		s.Len(printer.GetLines(), 0)
+		s.Len(printer.GetErrorLines(), 0)
+
+		// expect users not deleted
+		_, err = s.th.App.GetUser(s.th.BasicUser2.Id)
+		s.Require().Nil(err)
+	})
+
+	s.Run("Delete all user", func() {
```
Comment (mgdelacroix) [inline commands/user_e2e_test.go:177]: We should add a direct reference to local mode here to clarify the test case:

```suggestion
	s.Run("Delete all users through local mode", func() {
```

File: commands/user_e2e_test.go [hunk 5]
```diff
@@ -136,3 +136,64 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
+		printer.Clean()
+
+		// populate with some user
+		for i := 0; i < 10; i++ {
+			s.th.CreateUser()
```
Comment (mgdelacroix) [inline commands/user_e2e_test.go:182]: We should check the errors when creating the users:

```suggestion
			_, err := s.th.CreateUser()
			s.Require().Nil(err)
```

File: commands/user_e2e_test.go [hunk 6]
```diff
@@ -136,3 +136,64 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
+		cmd.Flags().BoolVar(&confirm, "confirm", confirm, "confirm")
+
+		err := deleteAllUsersCmdF(s.th.LocalClient, cmd, []string{})
+		s.Require().Nil(err)
+		s.Len(printer.GetLines(), 1)
+		s.Len(printer.GetErrorLines(), 0)
+		s.Require().Equal(printer.GetLines()[0], "All users successfully deleted")
+
+		// expect users deleted
+		_, err = s.th.App.GetUser(s.th.BasicUser.Id)
```
Comment (mgdelacroix) [inline commands/user_e2e_test.go:196]: Here we can use `s.th.App.GetUsers` instead of searching for a specific user to assert that the user list is empty

**Review by mgdelacroix — CHANGES_REQUESTED** [review]
Really nice work @M-Buntoro, thanks a lot for your contributions! Added a couple of minor request changes

File: commands/user_e2e_test.go (hunk 2 above)
Comment (M-Buntoro) [reply commands/user_e2e_test.go:172]: Oh I see, then the test should be okay
thanks @mgdelacroix for the confirmation

File: commands/user_e2e_test.go (hunk 4 above)
Comment (M-Buntoro) [reply commands/user_e2e_test.go]: Agree

File: commands/user_e2e_test.go (hunk 6 above)
Comment (M-Buntoro) [reply commands/user_e2e_test.go:197]: Ah, good point, let me improve that

Comment (M-Buntoro) [reply commands/user_e2e_test.go]: adjusted, please check again @mgdelacroix 
thanks

File: commands/user_e2e_test.go [hunk 7]
```diff
@@ -136,3 +136,83 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
+		// expect users not deleted
+		users, err := s.th.App.GetUsers(&model.UserGetOptions{
+			Page:    0,
+			PerPage: 10,
+		})
+		s.Require().Nil(err)
+		s.Require().NotZero(len(users))
+	})
+
+	s.Run("Delete all user as system admin", func() {
```
Comment (mgdelacroix) [inline commands/user_e2e_test.go:164]: nit: for clarity

```suggestion
	s.Run("Delete all user as system admin through the port API should not work", func() {
```

File: commands/user_e2e_test.go [hunk 8]
```diff
@@ -136,3 +136,83 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
 		s.Equal("Unable to find user '"+emailArg+"'", printer.GetErrorLines()[0])
 	})
 }
+
+func (s *MmctlE2ETestSuite) TestDeleteAllUserCmd() {
+	s.SetupTestHelper().InitBasic()
+
+	s.Run("Delete all user as unpriviliged user", func() {
```
Comment (mgdelacroix) [inline commands/user_e2e_test.go:143]: nit: for clarity

```suggestion
	s.Run("Delete all user as unpriviliged user should not work", func() {
```

File: commands/user_e2e_test.go [hunk 9]
```diff
@@ -136,3 +136,83 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
+		// expect users not deleted
+		users, err := s.th.App.GetUsers(&model.UserGetOptions{
+			Page:    0,
+			PerPage: 10,
+		})
+		s.Require().Nil(err)
+		s.Require().NotZero(len(users))
+	})
+
+	s.Run("Delete all users through local mode", func() {
```
Comment (mgdelacroix) [inline commands/user_e2e_test.go:185]: nit: for clarity

```suggestion
	s.Run("Delete all users through local mode should work correctly", func() {
```

**Review by mgdelacroix — APPROVED** [review]
LGTM! Thanks for the contribution @M-Buntoro! :confetti_ball: 

Added a couple of nonblocking suggestions for clarity on the test names

**Review by isacikgoz — APPROVED** [review]
Nicely done @M-Buntoro 🎉 And agree with @mgdelacroix reviews for clarity.

**Review by lindy65 — APPROVED** [review]
Thanks @M-Buntoro @mgdelacroix - looks good to me 👍

**Comment by hanzei** [comment]
Thanks for the PR @M-Buntoro :man_dancing:

**Pull request merged**